### PR TITLE
Add product tour to new product management experience

### LINKED ISSUE #36322: Intro modal for the new experience
When the new user enters the new experience for the first time, we should show them an intro modal.

**Modal**
`Title`: Meet the product editing form
`Body`: Let us show you how to navigate the form and create this product from start to finish in no time.
`Primary button`: Show me around (10 s)
`Secondary button`: I’ll explore on my own

`Desing`: lj4AUMJnpMtUe7XqyvOONq-fi-501%3A7172&t=7PveV6YsBsObVtis-1

If the user selects "Show me around (10 s)", we enable the spotlight tour with the five steps shown below.
Designs: lj4AUMJnpMtUe7XqyvOONq-fi-501%3A7372&t=npEAXmnjXwQbZ97A-0

- Step 1
`Design`: lj4AUMJnpMtUe7XqyvOONq-fi-501%3A7372&t=7PveV6YsBsObVtis-1
`Tooltip title`: 📣  Tell a story about your product
`Tooltip body`: The product form will help you describe your product field by field—from basic details like name and description to attributes the customers can use to find it on your store.

- Step 2
`Design`: lj4AUMJnpMtUe7XqyvOONq-fi-501%3A7248&t=7PveV6YsBsObVtis-1
`Tooltip title`: ✍️  Set up pricing & more
`Tooltip body`: When done, use the tabs to switch between other details and settings. In the future, you’ll also find here extensions and plugins.

- Step 3
`Design`: lj4AUMJnpMtUe7XqyvOONq-fi-501%3A7474&t=7PveV6YsBsObVtis-1
`Tooltip title`: 🔍  Preview and publish
`Tooltip body`: With all the details in place, use the buttons at the top to easily preview and publish your product. Click the arrow button for more options.

- Step 4
`Design`: lj4AUMJnpMtUe7XqyvOONq-fi-501%3A7614&t=7PveV6YsBsObVtis-1
`Tooltip title`: ⚙️  Looking for more?
`Tooltip body`: If the form doesn’t yet have all the features you need—it’s still in development—you can switch to the classic editor anytime.

- Step 5
`Design`: lj4AUMJnpMtUe7XqyvOONq-fi-501%3A7716&t=7PveV6YsBsObVtis-1
`Tooltip title`: 🛟  We’re here to help
`Tooltip body`: Check out the help panel for useful tips and links to step-by-step guides—or reach out to us with any questions.

**Bottom bar**
We will display [a bar at the bottom](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/customer-effort-score-tracks/product-mvp-ces-footer.tsx) that will allow the user to share feedback or disable the new experience. The bar will be shown if one of the following conditions is met:
- Closes the `Meet the product editing form` modal.
- Selects the "I’ll explore on my own" option in the modal
- After the user completes the spotlight tour.
- After the user closes the spotlight tour without finishing it.

The bar should be displayed permanently unless it's closed by the user.

Captura de ecrã 2022-10-17, às 16 29 58

**Acceptance criteria:**
- [ ] Show an intro modal when the new user enters the new experience for the first time.
- [ ] The modal is composed of the image, title, body, and buttons described above and in the design lj4AUMJnpMtUe7XqyvOONq-fi-501%3A7172&t=7PveV6YsBsObVtis-1
- [ ] If the user presses "Show me around (10 s)", we will enable the spotlight tour with the five steps described above.
- [ ] We will show the bottom bar if the user:
   - [ ] Closes the `Meet the product editing form` modal.
   - [ ] Selects the "I’ll explore on my own" option in the modal.
   - [ ] Completes the spotlight tour.
   - [ ] Closes the spotlight tour without finishing it.

### PR DESCRIPTION
**Pull request by joshuatf**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds the product tour modal and tour kit to the new product experience.

Screen Shot 2023-01-12 at 5 34 43 PM

Screen Shot 2023-01-13 at 10 06 23 AM

Closes (part of) #36322  .

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Navigate to the new product experience
2. See the modal appear after a few moments
3. Click the button to begin the tour
4. Walk through the tour, noting that copy and tooltip placement is correct
5. After completing the tour, delete `woocommerce_product_tour_modal_hidden` from your `wp_options` table
6. Reload the page and exit the tour early, making sure that the modal is dismissed
7. Make sure refreshing the page persists the hidden modal and that it does not appear again on page load

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/products/tour/test/product-tour-container.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+jest.mock( '../use-product-tour', () => {
+	return {
+		useProductTour: jest.fn(),
+	};
+} );
+
+const mockedUseProductTour = useProductTour as jest.Mock;
+
+describe( 'ProductTourContainer', () => {
```
Comment (joelclimbsthings) [inline plugins/woocommerce-admin/client/products/tour/test/product-tour-container.spec.tsx:33]: 🥳

File: plugins/woocommerce-admin/client/products/tour/product-tour.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+			{
+				referenceElements: {
+					desktop: `#tab-panel-0-pricing`,
+				},
+				meta: {
+					name: 'tabs',
+					heading: __( '✍️ Set up pricing & more', 'woocommerce' ),
+					descriptions: {
+						desktop: __(
+							'When done, use the tabs to switch between other details and settings. In the future, you’ll also find here extensions and plugins.',
```
Comment (joelclimbsthings) [inline plugins/woocommerce-admin/client/products/tour/product-tour.tsx:55]: Does the language with 
> In the future, you'll also **find here extensions and plugins**.

Feel weird to anyone else? Not sure if it's just me.

File: plugins/woocommerce-admin/client/products/tour/test/product-tour-container.spec.tsx (hunk 1 above)
Comment (joelclimbsthings) [inline plugins/woocommerce-admin/client/products/tour/test/product-tour-container.spec.tsx:33]: 🥳

**Review by joelclimbsthings — DISMISSED** [review]
Great job @joshuatf , this is testing well. I left an optional comment or two, and had a few other random thoughts:

- I can't help but feel that this might be a type of modal/tour combination that we may want to reuse with other features to keep the UI consistent as we introduce changes elsewhere. Don't have the need now of course, but this might be a good candidate for a general Tour+Modal component at some point.
- In terms of keyboard accessibility I noticed that the focus isn't brought to the first step after clicking "Show me around," although it is for subsequent steps. 
- Not really mobile friendly, but the MVP isn't in general at this point. Still this could be tech debt when we bring our attention to such things.

Nothing required above, so I'll preapprove.

File: plugins/woocommerce-admin/client/products/tour/product-tour.tsx (hunk 2 above)
Comment (joshuatf) [reply plugins/woocommerce-admin/client/products/tour/product-tour.tsx:55]: Definitely- good catch!  Updated to `In the future, you’ll also find extensions and plugins here.`. Sound okay to you @pmcpinto ?

**Comment by joshuatf** [comment]
Thanks for the feedback, @joelclimbsthings!

>I can't help but feel that this might be a type of modal/tour combination that we may want to reuse with other features to keep the UI consistent as we introduce changes elsewhere.

Totally agreed.  I really thought a component like this already existed and was surprised to not find one.  We should prioritize reusing this as much as possible the next opportunity we get.

>In terms of keyboard accessibility I noticed that the focus isn't brought to the first step after clicking "Show me around," although it is for subsequent steps.

Good point.  I feel the tour kit should probably handle focusing itself if no `focusElement` is provided for accessibility.  Any feelings on this @chihsuan or @rjchow?

>Not really mobile friendly, but the MVP isn't in general at this point. Still this could be tech debt when we bring our attention to such things.

Added in some very basic styling, hiding the tour on mobile since it's not as useful without spotlight, and some positioning tweaks on medium sized screens.  Let me know if you still want to show this on mobile and we can adjust @jarekmorawski.

**Comment by rjchow** [comment]
> Good point. I feel the tour kit should probably handle focusing itself if no focusElement is provided for accessibility

Makes sense to me!

**Comment by jarekmorawski** [comment]
On mobile, the spotlight message could show up at the bottom of the screen with the specific area highlighted, and the rest dimmed out. What do you think? I'd say it's nice to have depending on the amount of effort need to make it work.

image

**Comment by joshuatf** [comment]
>On mobile, the spotlight message could show up at the bottom of the screen with the specific area highlighted, and the rest dimmed out. What do you think? I'd say it's nice to have depending on the amount of effort need to make it work.

Definitely!  I think that would be a nice addition and seems possible.

Currently the spotlight seems to be turned off by the tour kit on mobile at the moment, so we'll need to follow up and modify that component to make this work on mobile.

Screen Shot 2023-01-17 at 9 46 38 AM
Notice is at bottom but no spotlight.

Screen Shot 2023-01-17 at 9 46 48 AM
Problematic when scrolled.

**Comment by jarekmorawski** [comment]
> Problematic when scrolled.

Maybe we could disable the scroll when the spotlight is displayed on the screen? Users could dismiss it or just sit back and enjoy the tour. 🙃

**Comment by joshuatf** [comment]
>Maybe we could disable the scroll when the spotlight is displayed on the screen? Users could dismiss it or just sit back and enjoy the tour. 🙃

Sounds like a good idea!  I'll open an issue to add in some of these changes to the tour components.

**Review by joelclimbsthings — APPROVED** [review]
Thanks for the follow-up @joshuatf , this is working well. It sounds like we have a couple other issues in mind around the Tour component?
- Fix focus state for first item
- Enable on mobile

That said, this looks good! 🚢

**Pull request merged**